pandas/core/groupby/groupby.py

```diff
@@ -899,10 +899,21 @@ def _python_agg_general(self, func, *args, **kwargs):
         output = {}
         for name, obj in self._iterate_slices():
             try:
-                result, counts = self.grouper.agg_series(obj, f)
+                # if this function is invalid for this dtype, we will ignore it.
```

Comment:
hmm, this makes us potentially run the function twice?

Reply:
its the same pattern we use in libreduction to call the function on any empty Series

Reply:
k cool, yeah I think we have some tests for this, interesting they didn't break